pinctrl/hw_pwm.go

```diff
@@ -205,68 +228,19 @@ func (pwm *pwmDevice) writeToPinModeByte(gpioNumber int, newMode byte) error {
 	*/
 
 	// Find base address of GPIO Pin in memory:
-	pinAddress, err := getGPIOPinAddress(gpioNumber)
+	pinAddress, err := getGPIOPinAddress(pwm.gpioLine)
 	if err != nil {
 		return fmt.Errorf("error getting gpio pin address: %w", err)
 	}
 
 	// Get pin's memory contents from virtual page; set the 5th byte of pin to mode
 	altModeIndex := int64(4)
 	vPage := *(pwm.gpioPinsPage)
-	vPage[pinAddress+altModeIndex] = newMode
+	vPage[pinAddress+altModeIndex] = pinMode
 
 	return nil
 }
 
-/*
-For all pins belonging to the same bank, pin data is stored contiguously and in 8 byte chunks.
-For a given pin, this method determines:
- 1. which bank the pin belongs to
- 2. the starting address of its 8 byte data chunk.
-*/
-func getGPIOPinAddress(gpioNumber int) (int64, error) {
-	const pinDataSizeBytes = 0x8 // 4 bytes = control status bits, 4 bytes to represent all possible control modes. 8 bytes per pin
-
-	if !(1 <= gpioNumber && gpioNumber <= numGPIOPins) {
-		return -1, errors.New("pin is out of bank range")
-	}
-
-	pinAddressOffset := (gpioNumber * pinDataSizeBytes)
-	return int64(pinAddressOffset), nil
-}
-
-/*
-This function uses pwm.line to determine what GPIO Pin it needs to set to the inputted mode
-Each pwm line corresponds to a GPIO Pin. The pi5 mapping is:
-
-	PWM Line 0 -> GPIO 12
-	PWM Line 1 -> GPIO 13
-	PWM Line 2 -> GPIO 18
-	PWM Line 3 -> GPIO 19
-
-Other mappings for different pis can be found here:  https://pypi.org/project/rpi-hardware-pwm/#modal-close
-
-Use the writeToPinModeByte to set the mode to PWM or GPIO using this helper method. Pin Mode will either be 'PWMMode' or 'GPIO'
-
-TODO: Make writing to GPIO Pins generalizeable, and not dependent on the pwm following this exact structure.
-*/
-func (pwm *pwmDevice) SetPinMode(pinMode byte) (err error) {
-	switch pwm.line {
-	case 0:
-		err = pwm.writeToPinModeByte(12, pinMode)
-	case 1:
-		err = pwm.writeToPinModeByte(13, pinMode)
-	case 2:
-		err = pwm.writeToPinModeByte(18, pinMode)
-	case 3:
-		err = pwm.writeToPinModeByte(19, pinMode)
```

Comment:
using the pinmapping defined in a `data.go` file for a board we realized we could get rid of this, but other parts of this file appear to reference pi5s still